test/apps/Makefile.am

```diff
@@ -122,9 +122,8 @@ endif
 .cu.o : ; $(NVCC) -c -o $@ $<
 
 test_cuda_get_symbol_address.o: test_cuda_get_symbol_address.cu
-	$(NVCC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(test_cuda_get_symbol_address_CPPFLAGS) -MT $@ -MD -MF \
-	$(DEPDIR)/test_cuda_get_symbol_address.Tpo -c -o $@ \
-	`test -f '$<' || echo '$(srcdir)/'`$<
-	$(AM_V_at)$(am__mv) $(DEPDIR)/test_cuda_get_symbol_address.Tpo \
-	$(DEPDIR)/test_cuda_get_symbol_address.Po
+@am__fastdepCC_TRUE@	$(AM_V_CC)$(NVCC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(test_cuda_get_symbol_address_CPPFLAGS) -Xcompiler "$(CPPFLAGS) $(CFLAGS)" -MT $@ -MD -MF $(DEPDIR)/test_cuda_get_symbol_address.Tpo -c -o $@  `test -f 'test_cuda_get_symbol_address.cu' || echo '$(srcdir)/'`test_cuda_get_symbol_address.cu
```

Comment:
any way to make this simpler? maybe some kind of intermediate/explicit dependency?
what is the issue being solved here?

Reply:
This is the same command line that is autogenerated and used to build objects from .c files. The only difference are: `$(NVCC)` instead of `$(CC)`, `-Xcompiler` to pass options not supported by nvcc, and the absence of `-MP` option.

`@am__fastdepCC_TRUE@`, `@AMDEP_TRUE@@am__fastdepCC_FALSE@`, `@am__fastdepCC_FALSE@` are used to comment out the unused lines depending on the build type.

Reply:
Or at least remove some parts of the command which are not essential?

Reply:
I did not find an easier way to use two different command line for different compilers. I can try to simplify the command line a little. However, I do not know any other ways to support different command lines for make/make dist/etc... other than using the same variables to comment out the lines.